### PR TITLE
Add mainstems to ref_gages

### PR DESCRIPTION
**Pull request by dblodgett-usgs**

### CODE REVIEW

File: pygeoapi/pygeoapi.config.yml [hunk 1]
```diff
@@ -299,11 +299,15 @@ resources:
               subjectOf:
                   "@id": schema:subjectOf
                   "@type": "@id"
+            - hyf: https://www.opengis.net/def/schema/hy_features/hyf
+              mainstem_uri:
+                "@id": hyf:flowpath
```
Comment (dblodgett-usgs) [inline pygeoapi/pygeoapi.config.yml:304]: This is a bit of a stretch, but I think this makes the most sense. 
image

We are using the flowpath relation from hydronetwork to HY_Flowpath. The other relation that could be used in linear element, but I think it's just as indirect and less semantically useful.

**Comment by dblodgett-usgs** [comment]
The gpkg in hydroshare now has a mainstem_uri field that contains the PID for the mainstem that the gage is on.

**Comment by dblodgett-usgs** [comment]
Some gages will have mainstem URIs
image

Some will not.
image

File: pygeoapi/pygeoapi.config.yml (hunk 1 above)
Comment (dblodgett-usgs) [reply pygeoapi/pygeoapi.config.yml:304]: A couple other ways to conceptualize this. 

We could say that ref_gages are HY_Hydrolocation features of type hydrometricStation. In which case, we would link to the mainstem as the "linearElement" of the HY_Hydrolocation's indirect position. 

The other association would be to skip the indirect position concepts and just use sosa:hasUltimateFeatureofInterest. 

My knee jerk is to use sosa for mainstems and linearelement for the actual reachcode/measure indirect position. Thoughts??

**Comment by ksonda** [comment]
Minor syntax thing in config, hyf url prefix needs trailing "/"

**Comment by dblodgett-usgs** [comment]
Should be good now -- Want to merge @ksonda? This would be a good one to try out the template scheme.

**Comment by ksonda** [comment]
closes #144

**Pull request merged**